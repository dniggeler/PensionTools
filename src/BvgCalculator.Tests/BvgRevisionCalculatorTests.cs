using System;
using System.Collections.Generic;
using Application.Bvg.Models;
using Domain.Enums;
using Domain.Models.Bvg;
using FluentAssertions;
using LanguageExt;
using Snapshooter.Xunit;
using Xunit;

namespace BvgCalculator.Tests;

[Trait("BVG", "Calculator")]
public class BvgRevisionCalculatorTests : IClassFixture<BvgCalculatorFixture<Application.Bvg.BvgRevisionCalculator>>
{
    private readonly BvgCalculatorFixture<Application.Bvg.BvgRevisionCalculator> _fixture;

    public BvgRevisionCalculatorTests(BvgCalculatorFixture<Application.Bvg.BvgRevisionCalculator> fixture)
    {
        _fixture = fixture;
    }

    [Fact(DisplayName = "BVG Result When Retiring")]
    public void ShouldCalculateResultWhenRetiring()
    {
        // given
        DateTime processDate = new DateTime(2024, 1, 1);
        DateTime birthdate = new DateTime(1959, 3, 13);
        decimal retirementCapitalEndOfYear = 0;

        BvgPerson person = _fixture.GetTestPerson(birthdate);
        person.ReportedSalary = 85000M;
        person.PartTimeDegree = 0.8M;

        // when
        Either<string, BvgCalculationResult> response = _fixture.GetBvgBenefits(retirementCapitalEndOfYear, person, processDate);

        BvgCalculationResult result = response.IfLeft(err => throw new ApplicationException(err));

        // then
        result.RetirementCapitalSequence.Should().NotBeNullOrEmpty();

        Snapshot.Match(result);
    }

    [Fact(DisplayName = "Default BVG Result")]
    public void ShouldCalculateDefaultResult()
    {
        // given
        DateTime processDate = new DateTime(2024, 1, 1);
        DateTime birthdate = new DateTime(1969, 12, 17);
        decimal retirementCapitalEndOfYear = 0;

        BvgPerson person = _fixture.GetTestPerson(birthdate);

        // when
        var response = _fixture.GetBvgBenefits(retirementCapitalEndOfYear, person, processDate);

        BvgCalculationResult result = response.IfLeft(err => throw new ApplicationException(err));

        // then
        result.RetirementCapitalSequence.Should().NotBeNullOrEmpty();

        Snapshot.Match(result);
    }

    [Fact(DisplayName = "Calculate Benefits")]
    public void ShouldReturnBenefitsCalculationResult()
    {
        // given
        DateTime processDate = new DateTime(2024, 1, 1);
        DateTime birthdate = new DateTime(1974, 8, 31);

        BvgPerson person = _fixture.GetTestPerson(birthdate);

        // when
        var response = _fixture.GetBvgBenefits(0, person, processDate);

        BvgCalculationResult result = response.IfLeft(err => throw new ApplicationException(err));

        // then
        Snapshot.Match(result);
    }

    [Fact(DisplayName = "Benefits If Below Salary Threshold")]
    public void ShouldReturnBenefitsIfBelowSalaryThreshold()
    {
        // given
        DateTime processDate = new DateTime(2024, 1, 1);

        BvgPerson person = _fixture.GetCurrentPersonDetails(new DateTime(1974, 8, 31), 20_000, 1M);

        // when
        var response = _fixture.GetBvgBenefits(0, person, processDate);

        BvgCalculationResult result = response.IfLeft(err => throw new ApplicationException(err));

        // then
        result.RetirementCapitalSequence.Should().NotBeNullOrEmpty();
        result.RetirementCreditSequence.Should().NotBeNullOrEmpty();

        Snapshot.Match(result);
    }

    [Theory(DisplayName = "BVG Benefits")]
    [MemberData(nameof(GetTestData))]
    public void Calculate_Bvg_Benefits(
        string dateOfProcessString,
        decimal salary,
        string dateOfBirthString,
        int genderCode,
        decimal currentRetirementCapital,
        decimal expectedRetirementPension)
    {
        DateTime dateOfProcess = DateTime.Parse(dateOfProcessString);
        DateTime dateOfBirth = DateTime.Parse(dateOfBirthString);
        Gender gender = (Gender)genderCode;

        BvgPerson person = _fixture.GetCurrentPersonDetails(dateOfBirth, salary, 1M);
        person.Gender = gender;

        var response = _fixture.GetBvgBenefits(currentRetirementCapital, person, dateOfProcess);

        BvgCalculationResult result = response.IfLeft(err => throw new ApplicationException(err));

        // then
        result.RetirementPension.Should().BeApproximately(expectedRetirementPension, 1M);
    }

    [Theory(DisplayName = "Insured Salary")]
    [InlineData("2024-01-01", 100_000, "1969-03-17", 1, 62475)]
    [InlineData("2024-01-01", 20_000, "1969-03-17", 1, 0)]
    [InlineData("2024-01-01", 18_000, "1969-03-17", 1, 0)]
    [InlineData("2026-01-01", 100_000, "1969-03-17", 1, 70560)]
    [InlineData("2026-01-01", 20_000, "1969-03-17", 1, 16000)]
    [InlineData("2026-01-01", 18_000, "1969-03-17", 1, 0)]
    public void Calculate_Insured_Salary(
        string dateOfProcessString,
        decimal effectiveSalary,
        string dateOfBirthString,
        int genderCode,
        decimal expectedInsuredSalary)
    {
        DateTime dateOfProcess = DateTime.Parse(dateOfProcessString);
        DateTime dateOfBirth = DateTime.Parse(dateOfBirthString);
        Gender gender = (Gender)genderCode;

        BvgPerson person = _fixture.GetCurrentPersonDetails(dateOfBirth, effectiveSalary, 1M);
        person.Gender = gender;

        Either<string, decimal> response = _fixture.Calculator().InsuredSalary(dateOfProcess, person);

        decimal result = response.IfLeft(err => throw new ApplicationException(err));

        // then
        result.Should().Be(expectedInsuredSalary);
    }

    [Fact(DisplayName = "Salary Array for BVG Maximum")]
    public void Calculate_Insured_Salaries_For_BVG_Maximum()
    {
        // given
        DateTime dateOfProcess = new(2024, 1, 1);
        DateTime dateOfBirth = new(1969, 12, 17);
        Gender gender = Gender.Male;
        decimal reportedSalary = 100_000;

        BvgPerson person = _fixture.GetCurrentPersonDetails(dateOfBirth, reportedSalary, 1M);
        person.Gender = gender;

        Either<string, BvgTimeSeriesPoint[]> response = _fixture.Calculator().InsuredSalaries(dateOfProcess, person);

        BvgTimeSeriesPoint[] result = response.IfLeft(err => throw new ApplicationException(err));

        // then
        result.Should().NotBeNullOrEmpty();
        Snapshot.Match(result);
    }

    [Fact(DisplayName = "Salary Array For Between Thresholds")]
    public void Calculate_Insured_Salaries_Between_Threshold()
    {
        // given
        DateTime dateOfProcess = new(2024, 1, 1);
        DateTime dateOfBirth = new(1974, 8, 31);
        Gender gender = Gender.Female;
        decimal reportedSalary = 20_000;

        BvgPerson person = _fixture.GetCurrentPersonDetails(dateOfBirth, reportedSalary, 1M);
        person.Gender = gender;

        Either<string, BvgTimeSeriesPoint[]> response = _fixture.Calculator().InsuredSalaries(dateOfProcess, person);

        BvgTimeSeriesPoint[] result = response.IfLeft(err => throw new ApplicationException(err));

        // then
        result.Should().NotBeNullOrEmpty();
        Snapshot.Match(result);
    }

    [Fact(DisplayName = "Retirement Credits Array")]
    public void Calculate_Retirement_Credits_Array()
    {
        // given
        DateTime dateOfProcess = new(2024, 1, 1);
        DateTime dateOfBirth = new(1969, 12, 17);
        Gender gender = Gender.Male;
        decimal reportedSalary = 100_000;

        BvgPerson person = _fixture.GetCurrentPersonDetails(dateOfBirth, reportedSalary, 1M);
        person.Gender = gender;

        Either<string, BvgTimeSeriesPoint[]> response = _fixture.Calculator().RetirementCredits(dateOfProcess, person);

        BvgTimeSeriesPoint[] result = response.IfLeft(err => throw new ApplicationException(err));

        // then
        result.Should().NotBeNullOrEmpty();
        Snapshot.Match(result);
    }

    [Fact(DisplayName = "Retirement Credit Factors Array")]
    public void Calculate_Retirement_Credit_Factors_Array()
    {
        // given
        DateTime dateOfProcess = new(2024, 1, 1);
        DateTime dateOfBirth = new(1969, 12, 17);
        Gender gender = Gender.Male;
        decimal reportedSalary = 100_000;

        BvgPerson person = _fixture.GetCurrentPersonDetails(dateOfBirth, reportedSalary, 1M);
        person.Gender = gender;

        Either<string, BvgTimeSeriesPoint[]> response = _fixture.Calculator().RetirementCreditFactors(dateOfProcess, person);

        BvgTimeSeriesPoint[] result = response.IfLeft(err => throw new ApplicationException(err));

        // then
        result.Should().NotBeNullOrEmpty();
        Snapshot.Match(result);
    }

    [Fact(DisplayName = "Retirement Credit Factors Array Young")]
    public void Calculate_Retirement_Credit_Factors_Array_Young()
    {
        // given
        DateTime dateOfProcess = new(2024, 1, 1);
        DateTime dateOfBirth = new(2002, 3, 17);
        Gender gender = Gender.Male;
        decimal reportedSalary = 50_000;

        BvgPerson person = _fixture.GetCurrentPersonDetails(dateOfBirth, reportedSalary, 1M);
        person.Gender = gender;

        Either<string, BvgTimeSeriesPoint[]> response = _fixture.Calculator().RetirementCreditFactors(dateOfProcess, person);

        BvgTimeSeriesPoint[] result = response.IfLeft(err => throw new ApplicationException(err));

        // then
        result.Should().NotBeNullOrEmpty();
        Snapshot.Match(result);
    }

    public static IEnumerable<object[]> GetTestData()
    {
        yield return
        [
            "2024-01-01",
            100_000,
            "1969-03-17",
            Gender.Male,
            0,
            6622
        ];
    }
}
