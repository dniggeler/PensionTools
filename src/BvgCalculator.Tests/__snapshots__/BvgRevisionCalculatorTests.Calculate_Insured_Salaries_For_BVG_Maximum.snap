[
  {
    "Age": {
      "Years": 54,
      "Months": 0
    },
    "Date": "2024-01-01T00:00:00",
    "Value": 62475.00
  },
  {
    "Age": {
      "Years": 54,
      "Months": 1
    },
    "Date": "2024-02-01T00:00:00",
    "Value": 62475.00
  },
  {
    "Age": {
      "Years": 54,
      "Months": 2
    },
    "Date": "2024-03-01T00:00:00",
    "Value": 62475.00
  },
  {
    "Age": {
      "Years": 54,
      "Months": 3
    },
    "Date": "2024-04-01T00:00:00",
    "Value": 62475.00
  },
  {
    "Age": {
      "Years": 54,
      "Months": 4
    },
    "Date": "2024-05-01T00:00:00",
    "Value": 62475.00
  },
  {
    "Age": {
      "Years": 54,
      "Months": 5
    },
    "Date": "2024-06-01T00:00:00",
    "Value": 62475.00
  },
  {
    "Age": {
      "Years": 54,
      "Months": 6
    },
    "Date": "2024-07-01T00:00:00",
    "Value": 62475.00
  },
  {
    "Age": {
      "Years": 54,
      "Months": 7
    },
    "Date": "2024-08-01T00:00:00",
    "Value": 62475.00
  },
  {
    "Age": {
      "Years": 54,
      "Months": 8
    },
    "Date": "2024-09-01T00:00:00",
    "Value": 62475.00
  },
  {
    "Age": {
      "Years": 54,
      "Months": 9
    },
    "Date": "2024-10-01T00:00:00",
    "Value": 62475.00
  },
  {
    "Age": {
      "Years": 54,
      "Months": 10
    },
    "Date": "2024-11-01T00:00:00",
    "Value": 62475.00
  },
  {
    "Age": {
      "Years": 54,
      "Months": 11
    },
    "Date": "2024-12-01T00:00:00",
    "Value": 62475.00
  },
  {
    "Age": {
      "Years": 55,
      "Months": 0
    },
    "Date": "2025-01-01T00:00:00",
    "Value": 62475.00
  },
  {
    "Age": {
      "Years": 55,
      "Months": 1
    },
    "Date": "2025-02-01T00:00:00",
    "Value": 62475.00
  },
  {
    "Age": {
      "Years": 55,
      "Months": 2
    },
    "Date": "2025-03-01T00:00:00",
    "Value": 62475.00
  },
  {
    "Age": {
      "Years": 55,
      "Months": 3
    },
    "Date": "2025-04-01T00:00:00",
    "Value": 62475.00
  },
  {
    "Age": {
      "Years": 55,
      "Months": 4
    },
    "Date": "2025-05-01T00:00:00",
    "Value": 62475.00
  },
  {
    "Age": {
      "Years": 55,
      "Months": 5
    },
    "Date": "2025-06-01T00:00:00",
    "Value": 62475.00
  },
  {
    "Age": {
      "Years": 55,
      "Months": 6
    },
    "Date": "2025-07-01T00:00:00",
    "Value": 62475.00
  },
  {
    "Age": {
      "Years": 55,
      "Months": 7
    },
    "Date": "2025-08-01T00:00:00",
    "Value": 62475.00
  },
  {
    "Age": {
      "Years": 55,
      "Months": 8
    },
    "Date": "2025-09-01T00:00:00",
    "Value": 62475.00
  },
  {
    "Age": {
      "Years": 55,
      "Months": 9
    },
    "Date": "2025-10-01T00:00:00",
    "Value": 62475.00
  },
  {
    "Age": {
      "Years": 55,
      "Months": 10
    },
    "Date": "2025-11-01T00:00:00",
    "Value": 62475.00
  },
  {
    "Age": {
      "Years": 55,
      "Months": 11
    },
    "Date": "2025-12-01T00:00:00",
    "Value": 62475.00
  },
  {
    "Age": {
      "Years": 56,
      "Months": 0
    },
    "Date": "2026-01-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 56,
      "Months": 1
    },
    "Date": "2026-02-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 56,
      "Months": 2
    },
    "Date": "2026-03-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 56,
      "Months": 3
    },
    "Date": "2026-04-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 56,
      "Months": 4
    },
    "Date": "2026-05-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 56,
      "Months": 5
    },
    "Date": "2026-06-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 56,
      "Months": 6
    },
    "Date": "2026-07-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 56,
      "Months": 7
    },
    "Date": "2026-08-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 56,
      "Months": 8
    },
    "Date": "2026-09-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 56,
      "Months": 9
    },
    "Date": "2026-10-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 56,
      "Months": 10
    },
    "Date": "2026-11-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 56,
      "Months": 11
    },
    "Date": "2026-12-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 57,
      "Months": 0
    },
    "Date": "2027-01-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 57,
      "Months": 1
    },
    "Date": "2027-02-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 57,
      "Months": 2
    },
    "Date": "2027-03-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 57,
      "Months": 3
    },
    "Date": "2027-04-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 57,
      "Months": 4
    },
    "Date": "2027-05-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 57,
      "Months": 5
    },
    "Date": "2027-06-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 57,
      "Months": 6
    },
    "Date": "2027-07-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 57,
      "Months": 7
    },
    "Date": "2027-08-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 57,
      "Months": 8
    },
    "Date": "2027-09-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 57,
      "Months": 9
    },
    "Date": "2027-10-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 57,
      "Months": 10
    },
    "Date": "2027-11-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 57,
      "Months": 11
    },
    "Date": "2027-12-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 58,
      "Months": 0
    },
    "Date": "2028-01-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 58,
      "Months": 1
    },
    "Date": "2028-02-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 58,
      "Months": 2
    },
    "Date": "2028-03-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 58,
      "Months": 3
    },
    "Date": "2028-04-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 58,
      "Months": 4
    },
    "Date": "2028-05-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 58,
      "Months": 5
    },
    "Date": "2028-06-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 58,
      "Months": 6
    },
    "Date": "2028-07-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 58,
      "Months": 7
    },
    "Date": "2028-08-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 58,
      "Months": 8
    },
    "Date": "2028-09-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 58,
      "Months": 9
    },
    "Date": "2028-10-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 58,
      "Months": 10
    },
    "Date": "2028-11-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 58,
      "Months": 11
    },
    "Date": "2028-12-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 59,
      "Months": 0
    },
    "Date": "2029-01-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 59,
      "Months": 1
    },
    "Date": "2029-02-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 59,
      "Months": 2
    },
    "Date": "2029-03-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 59,
      "Months": 3
    },
    "Date": "2029-04-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 59,
      "Months": 4
    },
    "Date": "2029-05-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 59,
      "Months": 5
    },
    "Date": "2029-06-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 59,
      "Months": 6
    },
    "Date": "2029-07-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 59,
      "Months": 7
    },
    "Date": "2029-08-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 59,
      "Months": 8
    },
    "Date": "2029-09-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 59,
      "Months": 9
    },
    "Date": "2029-10-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 59,
      "Months": 10
    },
    "Date": "2029-11-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 59,
      "Months": 11
    },
    "Date": "2029-12-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 60,
      "Months": 0
    },
    "Date": "2030-01-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 60,
      "Months": 1
    },
    "Date": "2030-02-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 60,
      "Months": 2
    },
    "Date": "2030-03-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 60,
      "Months": 3
    },
    "Date": "2030-04-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 60,
      "Months": 4
    },
    "Date": "2030-05-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 60,
      "Months": 5
    },
    "Date": "2030-06-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 60,
      "Months": 6
    },
    "Date": "2030-07-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 60,
      "Months": 7
    },
    "Date": "2030-08-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 60,
      "Months": 8
    },
    "Date": "2030-09-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 60,
      "Months": 9
    },
    "Date": "2030-10-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 60,
      "Months": 10
    },
    "Date": "2030-11-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 60,
      "Months": 11
    },
    "Date": "2030-12-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 61,
      "Months": 0
    },
    "Date": "2031-01-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 61,
      "Months": 1
    },
    "Date": "2031-02-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 61,
      "Months": 2
    },
    "Date": "2031-03-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 61,
      "Months": 3
    },
    "Date": "2031-04-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 61,
      "Months": 4
    },
    "Date": "2031-05-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 61,
      "Months": 5
    },
    "Date": "2031-06-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 61,
      "Months": 6
    },
    "Date": "2031-07-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 61,
      "Months": 7
    },
    "Date": "2031-08-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 61,
      "Months": 8
    },
    "Date": "2031-09-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 61,
      "Months": 9
    },
    "Date": "2031-10-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 61,
      "Months": 10
    },
    "Date": "2031-11-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 61,
      "Months": 11
    },
    "Date": "2031-12-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 62,
      "Months": 0
    },
    "Date": "2032-01-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 62,
      "Months": 1
    },
    "Date": "2032-02-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 62,
      "Months": 2
    },
    "Date": "2032-03-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 62,
      "Months": 3
    },
    "Date": "2032-04-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 62,
      "Months": 4
    },
    "Date": "2032-05-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 62,
      "Months": 5
    },
    "Date": "2032-06-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 62,
      "Months": 6
    },
    "Date": "2032-07-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 62,
      "Months": 7
    },
    "Date": "2032-08-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 62,
      "Months": 8
    },
    "Date": "2032-09-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 62,
      "Months": 9
    },
    "Date": "2032-10-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 62,
      "Months": 10
    },
    "Date": "2032-11-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 62,
      "Months": 11
    },
    "Date": "2032-12-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 63,
      "Months": 0
    },
    "Date": "2033-01-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 63,
      "Months": 1
    },
    "Date": "2033-02-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 63,
      "Months": 2
    },
    "Date": "2033-03-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 63,
      "Months": 3
    },
    "Date": "2033-04-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 63,
      "Months": 4
    },
    "Date": "2033-05-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 63,
      "Months": 5
    },
    "Date": "2033-06-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 63,
      "Months": 6
    },
    "Date": "2033-07-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 63,
      "Months": 7
    },
    "Date": "2033-08-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 63,
      "Months": 8
    },
    "Date": "2033-09-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 63,
      "Months": 9
    },
    "Date": "2033-10-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 63,
      "Months": 10
    },
    "Date": "2033-11-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 63,
      "Months": 11
    },
    "Date": "2033-12-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 64,
      "Months": 0
    },
    "Date": "2034-01-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 64,
      "Months": 1
    },
    "Date": "2034-02-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 64,
      "Months": 2
    },
    "Date": "2034-03-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 64,
      "Months": 3
    },
    "Date": "2034-04-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 64,
      "Months": 4
    },
    "Date": "2034-05-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 64,
      "Months": 5
    },
    "Date": "2034-06-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 64,
      "Months": 6
    },
    "Date": "2034-07-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 64,
      "Months": 7
    },
    "Date": "2034-08-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 64,
      "Months": 8
    },
    "Date": "2034-09-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 64,
      "Months": 9
    },
    "Date": "2034-10-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 64,
      "Months": 10
    },
    "Date": "2034-11-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 64,
      "Months": 11
    },
    "Date": "2034-12-01T00:00:00",
    "Value": 70560.00
  },
  {
    "Age": {
      "Years": 65,
      "Months": 0
    },
    "Date": "2035-01-01T00:00:00",
    "Value": 70560.00
  }
]
